src/main/java/de/tum/in/www1/artemis/service/compass/CompassCalculationEngine.java

```diff
@@ -432,16 +433,14 @@ private void addNewManualAssessment(List<Feedback> modelingAssessment, UMLDiagra
     }
 
     private boolean hasConflict(int elementId) {
-        Optional<Assessment> assessment = this.assessmentIndex.getAssessment(elementId);
-        if (assessment.isPresent()) {
-            for (List<Feedback> feedbacks : assessment.get().getContextFeedbackList().values()) {
-                double uniqueScore = -1;
-                for (Feedback feedback : feedbacks) {
-                    if (uniqueScore != -1 && uniqueScore != feedback.getCredits()) {
-                        return true;
-                    }
-                    uniqueScore = feedback.getCredits();
+        Optional<SimilaritySetAssessment> optionalAssessment = assessmentIndex.getAssessmentForSimilaritySet(elementId);
+        if (optionalAssessment.isPresent()) {
+            double uniqueScore = -1;
```

Comment:
I am wondering whether `-1` is a good `null` value here, as this could be a legit credit value. Maybe consider using `Double` instead with a proper `null` value.
```suggestion
            Double uniqueScore = null;
```

Reply:
Good catch, -1 is indeed a valid credit value. I simplified/refactored the method a bit, it now uses the stream API to check if all credits are the same in the feedback list. Therefore, your suggested change is obsolete now.